resource/cmake/CommonFunctions.cmake

```
@@ -0,0 +1,443 @@
+    option(RTIGATEWAY_ENABLE_EXAMPLES "Build examples applications for enabled plugins" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_DOCS "Build documentation for enabled plugins" OFF)
+    option(RTIGATEWAY_ENABLE_SSL "Enable support for SSL/TLS" OFF)
+    option(RTIGATEWAY_ENABLE_LOG "Enable logging to stdout" OFF)
+    option(RTIGATEWAY_ENABLE_TRACE "Enable support for trace-level logging" OFF)
+
+    m_init_install_path()
+
+    # Set RTIGATEWAY_CONFIGURE_CONNEXT to check whether we need to configure
+    # RTI Connext DDS.
+    set(RTIGATEWAY_CONFIGURE_CONNEXT OFF)
+    if (RTIGATEWAY_ENABLE_MODBUS
+            OR RTIGATEWAY_ENABLE_MQTT
+            OR RTIGATEWAY_ENABLE_FWD
+            OR RTIGATEWAY_ENABLE_TSFM_FIELD
+            OR RTIGATEWAY_ENABLE_TSFM_JSON)
+        set(RTIGATEWAY_CONFIGURE_CONNEXT ON)
+    endif()
+
+    if (RTIGATEWAY_ENABLE_TESTS)
+        enable_testing()
+    endif()
+
+endmacro()
+
+#[[
+m_configure_connext_dds
+---------------------
+
+ * Type: macro
+ * Brief: configures RTI Connext DDS by using its FindPackage
+ * Params:
+ ** _CONNEXTDDS_VERSION: version of RTI Connext DDS to configure
+ * Restrictions: CONNEXTDDS_DIR should be set
+ * How to use it:
+
+    m_configure_connext_dds(6.1.0)
+]]
+
+macro(m_configure_connext_dds _CONNEXTDDS_VERSION)
+    # Configure Connext DDS dependencies
+    set_required_variable(CONNEXTDDS_DIR NDDSHOME CONNEXTDDS_DIR)
+
+    # CONNEXTDDS_ARCH is not required, but recommended
+    # eg -DCONNEXTDDS_ARCH=x64Linux4gcc7.3.0
+    if (DEFINED CONNEXTDDS_ARCH)
+        message(STATUS "CONNEXTDDS_ARCH set to <${CONNEXTDDS_ARCH}>")
+    endif()
+
+    list(APPEND CMAKE_MODULE_PATH "${CONNEXTDDS_DIR}/resource/cmake")
+
+    find_package(
+        RTIConnextDDS
+            ${_CONNEXTDDS_VERSION}
+        COMPONENTS
+            routing_service core
+        REQUIRED)
+
+    message(STATUS "definition added ${CONNEXTDDS_DEFINITIONS}")
+
+    # Configure Connext DDS libraries variables
+    set(ROUTING_SERVICE_CXX_LIBRARIES
+        RTIConnextDDS::routing_service_cpp2
+    )
+    set(ROUTING_SERVICE_C_LIBRARIES
+        RTIConnextDDS::routing_service_c
+    )
+endmacro()
+
+#[[
+
+m_add_subdirectory_if_exists
+--------------------------
+
+ * Type: macro
+ * Brief: adds a subdirectory if it exists
+ * Params:
+ ** _SUBDIRECTORY_DIR: path to the subdirectory to add
+ * How to use it:
+
+    m_add_subdirectory_if_exists("${CMAKE_CURRENT_SOURCE_DIR}/mySubdirectory")
+]]
+macro(m_add_subdirectory_if_exists _SUBDIRECTORY_DIR)
+    if (IS_DIRECTORY "${_SUBDIRECTORY_DIR}")
+        add_subdirectory("${_SUBDIRECTORY_DIR}")
+    endif()
+endmacro()
+
+#[[
+
+m_add_doc_if_exist
+----------------
+
+ * Type: macro
+ * Brief: adds the "doc" directory if it exists in the current source
+          directory. This folder is only added if "building doc" is enabled.
+ * Params: no parameters
+ * How to use it:
+
+    m_add_doc_if_exist()
+]]
+macro(m_add_doc_if_exist)
```

I would rename this macro and the following one to make them shorter and (I think) a bit more specific:

```suggestion
macro(rtigw_add_docs)
```